tests/base/bounding_box_1.cc

```diff
@@ -104,6 +105,52 @@ test_bounding_box()
     deallog << "Boxes should not be equal : " << (b2 != b) << std::endl;
   }
   deallog << std::endl;
+
+  // Check if box is in an invalid state.
+  {
+    BoundingBox<spacedim> b;
+    b.get_boundary_points().first[0] = 1.0;
```

Comment:
This is why we should not have member functions that expose internal data. `get_boundary_points()` should not return a writable reference.

Reply:
I agree that it is not nice that one can mess around with the internal state and leave the class in an invalid state; but unfortunately this is how it is.

Reply:
Are we using this anywhere? Let me try what breaks if we return a `const` reference...

Reply:
It turns out to be not so difficult, see #14977. I would really like us see enforce the invariant that the constructor of `BoundingBox` already checks (you can't create an "invalid" bounding box) but that we current can't maintain because we expose the internal data structures in a writable way.

Reply:
This is required if you want to use our bounding box *as is* with boost, since the requirement for boost is that a you can modify a bounding box after you created one, and requires you to provide a writeable getter for the coordinates of the corner points.

Reply:
@luca-heltai Right, but one way to change it is by creating a new one and then assigning that to the old one. This allows us to check whether certain invariants are preserved, which we can't if we make internal data structures writable. #14977 shows how that works.

Reply:
I have removed the function so that we can proceed with the other additions. I have created a backup at: https://github.com/peterrum/dealii/tree/bb_invalid